pytest_watch/watcher.py

```diff
@@ -92,8 +94,8 @@ def run(self, summary=None):
             print(STYLE_NORMAL + msg + Fore.RESET + Style.NORMAL)
         if self.beforerun:
             os.system(self.beforerun)
-        exit_code = subprocess.call(['py.test'] + self.args,
-                                    shell=(sys.platform == 'win32'))
+        exit_code = pytest.main(self.args, plugins=[ReportCollectorPlugin()])
```

Comment:
Ok, I am taking another try with using subprocess, changing ReportCollectorPlugin into a  actual registered pytest plugin and using tempfile instead of env variable, whose path would then be set in env for onpass/onfail code.